features/dapp_develop/dapp_develop_test.go

```diff
@@ -108,8 +120,46 @@ func deployContract(ctx context.Context, compiledContractFileName string) error
 	testConfig := ctx.Value(e2e.TestConfigContextKey).(*testConfig)
 	contractWorkingDirectory := fmt.Sprintf("%s/soroban_examples", testConfig.TestWorkingDir)
 
+	var envCmd *cmd.Cmd
+
+	if testConfig.InstalledContractId != "" {
+		envCmd = cmd.NewCmd("/bin/sh", "-c",
```

Comment:
I'm not sure why you're running `sh` instead of invoking the process explicitly.

Reply:
I liked the readability a bit more, I'll revert back to command args for consistency

Reply:
I'm afraid that it's more error prone.. i.e it has search paths, it might fail to find the sh, etc

Reply:
yes, converted usage of `sh` to os exec instead, thanks - https://github.com/stellar/system-test/pull/8/commits/e89899c886138727aeef950e1210fdad3b53b5fd